### PR TITLE
[TechDocs] Update default techdocs docker image to latest version

### PR DESCRIPTION
**Pull request by emmaindal**
## Hey, I just made a Pull Request!

This PR sets the default techdocs docker image to the [latest released version](https://github.com/backstage/techdocs-container/releases/tag/0.3.3). It also adds a example to the docs of a newly introduced extension used for lists. 

**Before**
Screen Shot 2021-10-05 at 12 09 35 PM

**After**
Screen Shot 2021-10-05 at 12 27 52 PM

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by OrkoHunter — APPROVED** [review]

File: packages/techdocs-common/src/stages/generate/techdocs.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class TechdocsGenerator implements GeneratorBase {
    * The default docker image (and version) used to generate content. Public
    * and static so that techdocs-common consumers can use the same version.
    */
-  public static readonly defaultDockerImage = 'spotify/techdocs:v0.3.2';
+  public static readonly defaultDockerImage = 'spotify/techdocs:0.3.3';
```
Comment (OrkoHunter) [inline packages/techdocs-common/src/stages/generate/techdocs.ts:45]: Was omitting `v` from the tag intentional? Can cause confusion perhaps? :)
https://hub.docker.com/r/spotify/techdocs/tags?page=1&ordering=last_updated

**Review by camilaibs — APPROVED** [review]

File: packages/techdocs-common/src/stages/generate/techdocs.ts (hunk 1 above)
Comment (OrkoHunter) [reply packages/techdocs-common/src/stages/generate/techdocs.ts:45]: It's okay to omit it though and be consistent with that pattern for future.

Comment (OrkoHunter) [reply packages/techdocs-common/src/stages/generate/techdocs.ts:45]: Sorry for all the nitpicking, the name of the release has `v` in it, but the tag doesn't. https://github.com/backstage/techdocs-container/releases/tag/0.3.3

Comment (emmaindal) [reply packages/techdocs-common/src/stages/generate/techdocs.ts:45]: Oh yes that should be consistent. Thanks for pointing this out @OrkoHunter, I believe I can edit the tag, maybe I need to release again. Let me check!

Comment (emmaindal) [reply packages/techdocs-common/src/stages/generate/techdocs.ts]: Updated @OrkoHunter ✅

**Pull request merged**